### ISSUE TITLE
llvm: Cannot get llvm --with-clang

### ISSUE THREAD

**Issue by wreed4** [Homebrew/linuxbrew#382]
I've tried --HEAD, I've tried other versions of llvm (many of them have incorrect SHA values), I've grabbed the latest gcc (4.9).  But when I try to run "brew install llvm --with-clang", I always get to right after "make install" and then it appears to be trying to run clang, but can't find some headers.  

Any help here would be appreciated.  Thanks :)

**Comment by sjackman** [comment]
Please run `brew gist-logs llvm --with-clang` and post the gist URL

**Comment by wreed4** [comment]
@sjackman https://gist.github.com/46e19b4bcf464cf6db4e

**Comment by wreed4** [comment]
Saw that that last run asked me to do `brew update` so I did.  Here's another run:

https://gist.github.com/804b8e2499ac23d9d652

**Comment by sjackman** [comment]
This error makes it look like libc++ is being built for Mac. Is it possible to build libc++ for Linux?

https://gist.github.com/anonymous/804b8e2499ac23d9d652#file-04-make-L147-L150
```
+ /home/user/wreed/.linuxbrew/Cellar/llvm/HEAD/bin/clang algorithm.o bind.o chrono.o condition_variable.o debug.o exception.o future.o hash.o ios.o iostream.o locale.o memory.o mutex.o new.o optional.o random.o regex.o shared_mutex.o stdexcept.o string.o strstream.o system_error.o thread.o typeinfo.o utility.o valarray.o -arch i386 -arch x86_64 -o libc++.1.dylib -dynamiclib -nodefaultlibs -current_version 1 -compatibility_version 1 -install_name /usr/lib/libc++.1.dylib -lSystem -Wl,-unexported_symbols_list,libc++unexp.exp /usr/lib/libc++abi.dylib -Wl,-reexported_symbols_list,libc++abi.exp -Wl,-force_symbols_not_weak_list,notweak.exp -Wl,-force_symbols_weak_list,weak.exp -nostdinc++ -std=c++11 -fstrict-aliasing -Wall -Wextra -Wshadow -Wconversion -Wpadded -Wstrict-aliasing=2 -Wstrict-overflow=4
clang: error: no such file or directory: '/usr/lib/libc++abi.dylib'
make: *** [install] Error 1
make: Leaving directory `/tmp/llvm20150429-6626-1su4b6m/projects/libcxx'
```

**Comment by wreed4** [comment]
to my knowledge, it is....but I'm not sure.  I know it is possible to build llvm for linux.  Very difficult in my personal experience (hence wanting to use brew) but possible

**Comment by sjackman** [comment]
I'll guess that the default target is Darwin, and you have to pass some sort of configure option to target Linux.

**Comment by wreed4** [comment]
To the formula? Or do you mean the formula has to be edited?

**Comment by sjackman** [comment]
I mean the formula has to be edit to change how `libc++` is configured.

**Comment by aportelli** [comment]
Hi, I am working and struggling to solve this issue. Has anyone found a working solution yet?

**Comment by sjackman** [comment]
You can try scanning these articles:
+ https://solarianprogrammer.com/2013/01/17/building-clang-libcpp-ubuntu-linux/
+ http://stackoverflow.com/questions/25840088/how-to-build-libcxx-and-libcxxabi-by-clang-on-centos-7
+ http://stackoverflow.com/questions/17917731/what-is-the-sanctioned-way-to-build-libc-for-clang-on-linux

**Comment by aportelli** [comment]
Ok thanks!

**Comment by sjackman** [comment]
Please test the pull request #423.

**Comment by DoomHammer** [comment]
@aportelli @wreed4 Did the mentioned PR help?

**Comment by sjackman** [comment]
Fixed by #423.
